src/lib/annotations/drawing/DrawingController.js

```diff
@@ -0,0 +1,303 @@
+     * @param {AnnotationThread} thread - The thread that emitted the event
+     * @param {Object} data - Extra data related to the annotation event
+     * @return {void}
+     */
+    handleAnnotationEvent(thread, data = {}) {
+        switch (data.type) {
+            case 'locationassigned':
+                // Register the thread to the threadmap when a starting location is assigned. Should only occur once.
+                this.annotator.addThreadToMap(thread);
+                break;
+            case 'drawcommit':
+                // Upon a commit, remove the listeners on the thread.
+                // Adding the thread to the Rbush only happens upon a successful save
+                thread.removeAllListeners('annotationevent');
+                break;
+            case 'pagechanged':
+                // On page change, save the original thread, create a new thread and
+                // start drawing at the location indicating the page change
+                this.currentThread = undefined;
+                thread.saveAnnotation(TYPES.draw);
+                this.unbindModeListeners();
+                this.bindModeListeners(TYPES.draw);
+                this.currentThread.handleStart(data.location);
+                break;
+            case 'availableactions':
+                this.updateUndoRedoButtonEls(data.undo, data.redo);
+                break;
+            default:
+        }
+    }
+
+    /**
+     * Find the selected drawing threads given a pointer event. Randomly picks one if multiple drawings overlap
+     *
+     * @protected
+     * @param {Event} event - The event object containing the pointer information
+     * @return {void}
+     */
+    handleSelection(event) {
+        if (!event) {
+            return;
+        }
+
+        const location = this.annotator.getLocationFromEvent(event, TYPES.point);
+        if (!location) {
+            return;
+        }
+
+        const eventBoundary = {
+            minX: +location.x - DRAW_BORDER_OFFSET,
+            minY: +location.y - DRAW_BORDER_OFFSET,
+            maxX: +location.x + DRAW_BORDER_OFFSET,
+            maxY: +location.y + DRAW_BORDER_OFFSET
+        };
+
+        // Get the threads that correspond to the point that was clicked on
+        const intersectingThreads = this.threads
+            .search(eventBoundary)
+            .filter((drawingThread) => drawingThread.location.page === location.page);
+
+        // Clear boundary on previously selected thread
+        if (this.selected) {
+            const canvas = this.selected.drawingContext.canvas;
+            this.selected.drawingContext.clearRect(0, 0, canvas.width, canvas.height);
+        }
+
+        // Selected a region with no drawing threads, remove the reference to the previously selected thread
+        if (intersectingThreads.length === 0) {
+            this.selected = undefined;
+            return;
+        }
+
+        // Randomly select a thread in case there are multiple
```

Comment:
o_0

Reply:
This will change with better hit detection 👍  (using a hitmap canvas)